### DISCUSSION TITLE
Patch to fix orphan file when using db with OCC protocol need to be merged

### DISCUSSION THREAD

**Discussion by zhangh43** [juicedata/juicefs#4280] [General]
Hi all,

Not sure whether it's the right place to comment our requirement, but we do suffer from this issue.

When we use OCC protocol database as meta engine, there is a bug which leads to orphan file when conflict happens.
It is fixed by commit  on month ago.
```
https://github.com/juicedata/juicefs/commit/95ec414dcc2505b9c2a60324485cb562f24d2e8e

Return remove dir error of table dirStats to trigger retry. (https://github.com/juicedata/juicefs/pull/4134)
The current issue is that when there's an error in deleting dirstats,
the rmdir operation still proceeds (the transaction is successfully
committed). However, the next time a mkdir operation with the same name
is attempted, it will check dirstats. If it already exists, an error is
reported, making it impossible to make a directory with the same name.
```

But this commit (one month ago) is not merged to either v1.1.x or the current 1.2.0-dev branch. Do we have a plan to merge this commit to the next release tarball?

Thank you

**Comment by SandyXSD** [comment]
It IS contained in the current main (1.2.0-dev) branch, see: https://github.com/juicedata/juicefs/pull/4134

**Comment by davies** [comment]
The reason we did not include the #4134 in 1.1.1 is that we don't see how it could fail future mkdir(), since `dirStats` use inode as the unique key, and the inode of newly created dir should not be used before.  Can you provide steps to reproduce that issue?

**Comment by zhangh43** [comment]
> It IS contained in the current main (1.2.0-dev) branch, see: #4134

FYI It's in main, but not in tag 1.2.0-dev.

**Comment by davies** [comment]
tag 1.2.0-dev is not a release, just a tag for the beginning of v1.2.0

**Comment by zhangh43** [comment]
Thank you for notice
